app/assets/javascripts/discourse/app/lib/hashtag-autocomplete.js

```diff
@@ -187,8 +190,10 @@ function _searchGeneric(term, siteSettings, contextualHashtagConfiguration) {
           resolve(CANCELLED_STATUS);
         }, 5000);
 
-    if (term === "") {
-      return resolve(CANCELLED_STATUS);
+    if (!siteSettings.enable_experimental_hashtag_autocomplete) {
+      if (term === "") {
+        return resolve(CANCELLED_STATUS);
+      }
```

Comment:
```suggestion
    if (!siteSettings.enable_experimental_hashtag_autocomplete && term === "") {
      return resolve(CANCELLED_STATUS);
```

Minor but removes one level of nesting.